okhttp/src/main/kotlin/okhttp3/Dispatcher.kt

```diff
@@ -82,29 +88,31 @@ class Dispatcher() {
    * This means that if you are doing synchronous calls the network layer will not truly be idle
    * until every returned [Response] has been closed.
    */
-  @set:Synchronized
-  @get:Synchronized
   var idleCallback: Runnable? = null
+    get() = this.withLock { field }
```

Comment:
Maybe for follow up: should we do a property delegate so we can have syntax like this?
```
var idleCallback: Runnable? by lock.property(null)
```

(or similar)